torch_xla/csrc/init_python_bindings.cpp

```diff
@@ -1728,7 +1728,10 @@ void InitXlaModuleBindings(py::module m) {
             -> std::vector<at::Tensor> {
           XLA_CHECK(hash_str.size() == sizeof(torch::lazy::hash_t));
           torch::lazy::hash_t hash = *(torch::lazy::hash_t*)(hash_str.c_str());
-          torch::lazy::BackendDevice device = torch_xla::GetCurrentDevice();
+          // Device will be Virtual device if SPMD is enabled.
+          torch::lazy::BackendDevice device =
+              ShardingUtil::UseVirtualDevice() ? ParseDeviceString("SPMD:0")
+                                               : torch_xla::GetCurrentDevice();
```

Comment:
@yeounoh I am not sure if we should just update `GetCurrentDevice`, any thought? We need to sit down and think about how to surface this virtual device to user soon..

Reply:
I voted for GetCurrentDevice as there might be other scenario where the caller will also need to distinguish SPMD:0 with XLA:0.

Reply:
`GetCurrentDevice` is being used over 30 places in our code base now, mostly during tracing and caller trying to figure out the hw type. I think it should be fine as long as `SPMD:0` can be resolved into correct hardware type. I would leave that in a separate pr since it touches too many codes and might introduce noise.

Reply:
Got it.